batch/batch/api.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, timeout=60):
         self.timeout = timeout
 
     def create_job(self, url, spec, attributes, batch_id, callback, parent_ids,
-                   scratch_folder, input_files, output_files):
+                   scratch_folder, input_files, output_files, copy_service_account_name):
```

Comment:
So I feel like this isn't quite the right direction, although if you want it as an intermediate step, that's fine.

Obviously the user can't get to specify their service account.  It needs to come from their authenticated credentials.  I haven't looked at the final authn design, but the user account needs to come from claims in the cookie so batch either needs to look at the cookie (or headers created by authn after verifying the cookie).

Reply:
Yeah, I'm punting on proper authentication. The job spec is mostly unaudited, the user could specify any volume mounts or service accounts for the main pod they like. I'm just continuing to be embarrassingly permissive.

When we figure out authentication, we'll have the tool to also lock down the job spec.

My current short term thinking is to issue a pair of known random 64 character strings to Konrad and Liam which will serve as their passwords. I think that cribs us until we figure out a proper authentication flow involving oauth